taskfile/node_git.go

```diff
@@ -34,37 +34,43 @@ func NewGitNode(
 	insecure bool,
 	opts ...NodeOption,
 ) (*GitNode, error) {
-	base := NewBaseNode(dir, opts...)
-	u, err := giturls.Parse(entrypoint)
+	gitURL, err := giturls.Parse(entrypoint)
 	if err != nil {
 		return nil, err
 	}
+	if gitURL.Scheme == "http" && !insecure {
+		return nil, &errors.TaskfileNotSecureError{URI: entrypoint}
+	}
 
-	basePath, path := func() (string, string) {
-		x := strings.Split(u.Path, "//")
+	urlPath, filepath := func() (string, string) {
+		x := strings.Split(gitURL.Path, "//")
 		return x[0], x[1]
 	}()
-	ref := u.Query().Get("ref")
 
-	rawUrl := u.String()
+	ref := gitURL.Query().Get("ref")
+	rawURL := gitURL.String()
+	redactedRawURL := gitURL.Redacted()
 
-	u.RawQuery = ""
-	u.Path = basePath
+	gitURL.RawQuery = ""
+	gitURL.Path = urlPath
+	baseURL := gitURL.String()
 
-	if u.Scheme == "http" && !insecure {
-		return nil, &errors.TaskfileNotSecureError{URI: entrypoint}
-	}
 	return &GitNode{
-		BaseNode: base,
-		URL:      u,
-		rawUrl:   rawUrl,
-		ref:      ref,
-		path:     path,
+		BaseNode:       NewBaseNode(dir, opts...),
+		rawURL:         rawURL,
+		redactedRawURL: redactedRawURL,
+		baseURL:        baseURL,
+		ref:            ref,
+		filepath:       filepath,
 	}, nil
 }
 
 func (node *GitNode) Location() string {
-	return node.rawUrl
+	return node.rawURL
+}
+
+func (node *GitNode) RedactedLocation() string {
+	return node.redactedRawURL
```

Comment:
Using `node.URL.Redacted()` didn't work because it contained only the base URL without the Taskfile file path.